lib/Runtime/ByteCode/ScopeInfo.cpp

```diff
@@ -112,8 +112,17 @@ namespace Js
     ScopeInfo * ScopeInfo::SaveScopeInfo(ByteCodeGenerator* byteCodeGenerator, Scope * scope, ScriptContext * scriptContext)
     {
         // Advance past scopes that will be excluded from the closure environment. (But note that we always want the body scope.)
-        while (scope && (!scope->GetMustInstantiate() && scope != scope->GetFunc()->GetBodyScope()))
```

Comment:
Hm. So if the param scope is not going to be instantiated, that means there's nothing in that will be bound to anything. Why do we need to record these symbols that no one will bind to?

Reply:
Because we no longer put the body scope of the parent function in the scope chain. We would have always stopped and saved it before. Now the param scope can be in this chain in place of the body scope. If there's no scope from the parent function in the scope chain, we'll get confused if we did stack func and the nested function throws an error.